### PR TITLE
feat(host): Adds support for batch support to the host

### PR DESCRIPTION
**Pull request by thomastaylor312**
This enables keyvalue batch support inside of the host, along with a test to make sure it works. Not all of our providers implement batch yet, so this uses the Redis provider, which did have implementions. I did have to fix the redis provider to get the right type of data back and transform it. I also had to update our wRPC versions so we could pick up on some bug fixes for the types we are encoding in the batch interface.

### CODE REVIEW

File: src/bin/keyvalue-redis-provider/wasmcloud.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "KV Redis"
 language = "rust"
 type = "provider"
-version = "0.27.0"
+version = "0.28.0"
```
Comment (thomastaylor312) [inline src/bin/keyvalue-redis-provider/wasmcloud.toml:4]: The cargo.toml wasn't bumped because we've never released a 0.28

File: crates/provider-keyvalue-redis/src/lib.rs [hunk 2]
```diff
@@ -276,7 +276,20 @@ impl keyvalue::batch::Handler<Option<Context>> for KvRedisProvider {
         keys: Vec<String>,
     ) -> anyhow::Result<Result<Vec<Option<(String, Bytes)>>>> {
         check_bucket_name(&bucket);
-        Ok(self.exec_cmd(ctx, &mut Cmd::mget(&keys)).await)
+        let data = match self
```
Comment (rvolosatovs) [inline crates/provider-keyvalue-redis/src/lib.rs:279]: why the change here?

File: crates/runtime/src/component/keyvalue.rs [hunk 3]
```diff
@@ -66,6 +66,88 @@ where
...
+            &self.handler,
+            Some(ReplacedInstanceTarget::KeyvalueBatch),
+            bucket,
+            &keys,
+        )
+        .await?
+        {
+            Ok(res) => Ok(Ok(res
+                .into_iter()
+                .map(|opt| opt.map(|(k, v)| (k.to_string(), Vec::from(v))))
```
Comment (rvolosatovs) [inline crates/runtime/src/component/keyvalue.rs:95]:
```suggestion
                .map(|opt| opt.map(|(k, v)| (k, Vec::from(v))))
```

isn't `k` a `String` already?

**Review by rvolosatovs — APPROVED** [review]

File: crates/provider-keyvalue-redis/src/lib.rs (hunk 2 above)
Comment (thomastaylor312) [reply crates/provider-keyvalue-redis/src/lib.rs:279]: Because you have to massage the return data and map it to the Vec of tuples. So this fetches the data and then zips it up with the keys that were passed in

**Review by brooksmtownsend — APPROVED** [review]
